src/freenet/clients/http/SecurityLevelsToadlet.java

```diff
@@ -686,6 +693,25 @@ public static void generatePasswordFormPage(boolean wasWrong, ToadletContainer c
 		HTMLNode form = ctx.addFormChild(content, postTo, "masterPasswordForm");
 		generatePasswordFormPage(wasWrong, form, content, forDowngrade, forUpgrade, physicalSecurityLevel, redirect);
 	}
+	
+	public static void generatePasswordConfirmationForm(HTMLNode formNode) {
```

Comment:
There's no need to use this in a static context, it is only used (and to be used) by instances of `SecurityLevelsToadlet`. It's generally a good idea to reduce the exposure of any piece of code wherever possible.